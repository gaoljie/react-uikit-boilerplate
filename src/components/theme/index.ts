import React from "react";

export const theme = {
  primary: "#EF6F35",
  primaryFont: "Montserrat",
  secondaryFont: "Work Sans",
  desktop: "@media (max-width: 1920px)",
  smallDesktop: "@media (max-width: 1441px)",
  tablet: "@media (max-width: 769px)",
  mobile: "@media (max-width: 600px)"
};

export const ThemeContext = React.createContext(theme);

export const ThemeProvider = ThemeContext.Provider;
