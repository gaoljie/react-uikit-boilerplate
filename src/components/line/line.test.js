import React from "react";
import "jest-styled-components";
import { mount } from "enzyme";

import Line from "./index";
import theme from "../theme";

const props = {
  height: "5px",
  color: "black"
};

describe("Test Line", () => {
  it("Matches the snapshot", () => {
    const wrapper = mount(<Line {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Styling", () => {
    const wrapper = mount(<Line />);

    expect(wrapper).toHaveStyleRule("height", "2px");
    expect(wrapper).toHaveStyleRule("background-color", theme.primary);

    wrapper.setProps({ height: "3px", color: "black" });

    expect(wrapper).toHaveStyleRule("height", "3px");
    expect(wrapper).toHaveStyleRule("background-color", "black");
  });
});
