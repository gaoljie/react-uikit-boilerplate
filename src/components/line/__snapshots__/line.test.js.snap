// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Line Matches the snapshot 1`] = `
.c0 {
  height: 5px;
  color: black;
  background-color: black;
  height: 5px;
}

<Styled(styled.div)
  color="black"
  height="5px"
>
  <StyledComponent
    color="black"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            [Function],
            "
",
            "
  background-color: ",
            [Function],
            ";
  height: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    height="5px"
  >
    <div
      className="c0"
      color="black"
      height="5px"
    />
  </StyledComponent>
</Styled(styled.div)>
`;
